src/wp-admin/includes/post.php

```diff
@@ -1961,15 +1961,75 @@ function wp_create_post_autosave( $post_data ) {
 		 * @param array $new_autosave Post array - the autosave that is about to be saved.
 		 */
 		do_action( 'wp_creating_autosave', $new_autosave );
-
+		wp_autosave_post_revisioned_meta_fields( $new_autosave );
 		return wp_update_post( $new_autosave );
 	}
 
 	// _wp_put_post_revision() expects unescaped.
 	$post_data = wp_unslash( $post_data );
 
 	// Otherwise create the new autosave as a special post revision.
-	return _wp_put_post_revision( $post_data, true );
+	$revision = _wp_put_post_revision( $post_data, true );
+
+	// Update the revisioned meta data as well.
+	wp_autosave_post_revisioned_meta_fields( get_post( $revision, ARRAY_A ) );
+	return $revision;
+}
+
+/**
+ * Autosave the revisioned meta fields.
+ *
+ * Iterates thru the revisioned meta fields and checks each to see if they are set,
+ * and have a changed value. If so, the meta value is saved and attached to the autosave.
+ *
+ * @since 6.4.0
+ *
+ * @param array $new_autosave The new post data being autosaved.
+ */
+function wp_autosave_post_revisioned_meta_fields( $new_autosave ) {
```

Comment:
Why does this need to be a array and not a WP_Post object?

Reply:
The `wp_creating_autosave` hook where this is now hooked passes a post array. previously I was calling with that array and would have needed to transform into a post object.

Reply:
Is there some reason it would be preferable as a Post object?

Reply:
Just interested.